### DISCUSSION TITLE
Remove Sidebar

### DISCUSSION THREAD

**Discussion by 25233Guyver** [aclap-dev/video-downloadhelper#840] [Help and Q&A] [answered]
### Your question

I am not a fan of the sidebar opening on browser start up and I cannot find out how to disable it.

### Addon & CoApp version and details

version: 9.0.2.3
target: mozilla
channel: stable
lang: en-US
coapp: {"found":true,"path":"C:\\Program Files\\DownloadHelper CoApp\\vdhcoapp.exe","version":"2.0.19","new_version":false}
license: {"accepted":true,"email":"<removed>","key":"fec46c9d79184b3a"}
platform: x86-64 win
UA: Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:126.0) Gecko/20100101 Firefox/126.0
{
    "valid_license_message_has_been_displayed": true,
    "used_sidebar_button": true,
    "used_popup_button": true,
    "download_directory": "C:\\Users\\Zenme\\Desktop",
    "last_download_directory": {
        "__serializer_tag": "option_some",
        "__serializer_value": {
            "__serializer_tag": "primitive",
            "__serializer_value": "C:\\Users\\Zenme\\Desktop"
        }
    },
    "theme": "dark",
    "view_options": {
        "all_tabs": false,
        "low_quality": false,
        "sort_by_status": true,
        "sort_reverse": false,
        "show_button_clean": true,
        "show_button_clean_all": false,
        "show_button_convert_local": false,
        "hide_downloaded": false
    },
    "use_sidebar": false,
    "open_count_store": 5,
    "successfull_dl": 3
}

**Comment by rihntintin** [comment]
My sidebar is on the right side of the browser. At the bottom of the sidebar are 3 icons. Click on the middle icon. Done.

**Reply by 25233Guyver** [reply]
Let me rephrase: I don't want it happening at all.

**Reply by GrampaWildWilly** [reply]
I do not work for VDH.  I'm just another user.

As @rihntintin just told you, click the indicated control in the sidebar.  VDH will go back to always being a drop down menu instead.  But you can't remove the feature from the product.  If you don't select the sidebar, VDH won't show you the sidebar.  Just avoid that control.  You can't remove it.  You can only avoid it.

**Reply by 25233Guyver** [reply]
So far that actually hasn't worked for me. Every time I open up Firefox for the day, it's always there at start. Clicking on it will hide it, but the setting never seems to change.

**Reply by GrampaWildWilly** [reply]
Interesting.  That's not my experience with Windows 11 Pro 64-bit version 23H2 build 22631.3672, Firefox 126.0.1 64-bit, licensed VDH 9.0.2.3a1 beta, CoApp 2.0.19.  In order to remember things like settings & configurations, Firefox needs permissions to access relevant directories on your hard drive.  I would imagine you'd be seeing other problems in addition to this one if you were having permission problems.  Another possibility is the cleanup you do at browser shutdown.  It might be removing your settings changes.  Try this.  Pick some other VDH setting, like say Show low quality media.  That's just a check mark.  Change whatever you have then close all browser windows.  Then relaunch Firefox & see if that setting has persisted.

**Reply by GrampaWildWilly** [reply]
Seems to me I've also seen mention that certain antivirus settings can decide settings changes in Firefox are evidence of virus behavior so the antivirus undoes the changes.  I'm not sure how that ends up being set nor how to undo it.  I'm using the built in Windows Defender antivirus & it's not doing this to me.  There's an interesting comment in the Wiki about antivirus.

**Reply by 25233Guyver** [reply]
I am using Windows embedded as well, so we can rule that out I think.

**Reply by paulrouget** [reply]
> So far that actually hasn't worked for me. Every time I open up Firefox for the day, it's always there at start. Clicking on it will hide it, but the setting never seems to change.

This only happens on Firefox **startup**, correct?

I am pretty sure that a Firefox bug. I have talked to them about this, and iirc, they wrote the patch, but it hasn't been fixed in Firefox yet: https://phabricator.services.mozilla.com/D208352

**Reply by GrampaWildWilly** [reply]
That's weird.  A bug in Firefox?  How come @25233Guyver is seeing the problem but I'm not?

**Reply by 25233Guyver** [reply, marked as answer]
Alrighty, found the problem.

Me.

After restarting my computer about seven times and such, that's when I realized I was looking for the settings I was activating the side bar to get there and repeatedly didn't click to hide. Huzzah, user idiocy.

**Reply by GrampaWildWilly** [reply]
Been there done that.  😃

**Reply by Eldon-Tyrell** [reply]
I have the exact same problem (see [https://github.com/aclap-dev/video-downloadhelper/discussions/952#discussion-6884831](https://github.com/aclap-dev/video-downloadhelper/discussions/952#discussion-6884831))

and cannot fix it.

@25233Guyver
> . . . I was activating the side bar to get there and repeatedly didn't click to hide. . .

I don't quite get that - could you describe step by step what i have to do exactly to solve this ?

**Reply by 25233Guyver** [reply]
![image](https://github.com/aclap-dev/video-downloadhelper/assets/172090676/0d164bdb-2e47-43b2-a81e-7a1e0851b593)
Clicking this, and not looking for it anywhere else, will keep it to the toolbar.

**Reply by Eldon-Tyrell** [reply]
ah yes but that doesn't solve the problem of the bookmark-sidebar always being open on Firefox-start. - i thought you had a solution

**Reply by GrampaWildWilly** [reply]
My experience has been that you have to close the history and/or bookmarks sidebar in every Firefox window, then close all Firefox windows.  I believe it's possible to have different sidebars open in multiple Firefox windows/tabs.  It's also possible to have sidebars in some windows/tabs & not in others.  So you have to close all sidebars individually before you close your Firefox windows/tabs.  Once you've got everything well & truly closed, next time you open a Firefox window, you're supposed to not have any sidebar.  That doesn't really have anything to do with VDH.  At least, that's my understanding of how it's all supposed to work.  Is that not how you're seeing things?

**Reply by GrampaWildWilly** [reply]
I've just done a little experimenting.  It seems that within any given Firefox window, if you have multiple tabs open, they will either all have a bookmarks sidebar, or all have a history sidebar, or all have no sidebar.  Opening or closing or changing from one sidebar to the other is reflected to all the other tabs in that window.  But different Firefox windows can have different sidebars, or none, independently.  So you just have to be sure the last tab of the last open Firefox window has no sidebar & when you close that last one, that's supposed to tell Firefox to not open a sidebar next time you start Firefox.  Supposed to.  Are you seeing something different?

**Reply by GrampaWildWilly** [reply]
I thought maybe you were unknowingly opening Firefox with a switch that might be causing your problem.  I've just done a web search & found this: https://wiki.mozilla.org/Firefox/CommandLineOptions.  The last update date at the bottom of that page says 2024/5/27.  I thought maybe there was a switch for launching the browser with a particular sidebar.  There doesn't seem to be such a switch.  This appears to be an older version of the same information: https://www-archive.mozilla.org/docs/command-line-args.  There's no switch for sidebar in there, either.  This is quite baffling.

**Comment by 25233Guyver** [comment]
That was the solution actually. My dumbass was opening it again over and over in the side. It stopped as soon as I didn't try to click anything after that.
    On Tuesday, July 2, 2024 at 09:59:07 AM PDT, GrampaWildWilly ***@***.***> wrote:  
 
 


I thought maybe you were unknowingly opening Firefox with a switch that might be causing your problem. I've just done a web search & found this: https://wiki.mozilla.org/Firefox/CommandLineOptions. I thought maybe there was a switch for launching the browser with a particular sidebar. There doesn't seem to be such a switch. This appears to be an older version of the same information: https://www-archive.mozilla.org/docs/command-line-args. There's no switch for sidebar in there, either. This is quite baffling.

**Comment by Eldon-Tyrell** [comment]
i cannot follow you - please bear with me.

my problem: since I used the new VDH-UI once, my Firefox is in this state:

- my firefox shows the bookmark sidebar now.
- i close the sidebar by clicking the 'X'
- I close Firefox
- I start Firefox again
- the Bookmark-sidebar is there at Firefox start

always 
no matter what was there when closing Firefox (history, Bookmarks, VDH)
no matter if VDH is set to new or old UI
no matter if VDH is set to using the sidebar or pop-up 

I tried all combinations and orders I can imagine.

**Reply by 25233Guyver** [reply]
Press the button at the bottom of the sidebar, the one I showed in the previous image. Closing it using the X button will only close it but not prevent it from happening again.

**Reply by GrampaWildWilly** [reply]
Let me get this straight.

- You start with no Firefox windows open.
- You open a single Firefox window.
- It has the bookmarks sidebar open.
- You close the bookmarks sidebar.  When I do this, I use the keyboard shortcut Ctrl+b.  That is supposed to be the same as mouse clicking on the X.  But who knows?  Maybe it's different.  Maybe you can try that.
- However you close the bookmarks sidebar, next you close that solitary open Firefox window, which at this moment does NOT have any sidebar visible, not bookmarks, not history, not VDH.
- Next time you open Firefox, the bookmarks sidebar is open.

Do I have this right?  And this all started happening after you opened the VDH sidebar just one time, is that right?

**Reply by GrampaWildWilly** [reply]
I decided to hunt down some documentation.  You know, RTFM.  A dangerous, risky activity at the best of times.  The page you can see in the image below is this:

https://support.mozilla.org/en-US/kb/use-firefox-sidebar-access-bookmarks-history-synced

I decided to follow the instructions on that page & customize my Firefox toolbar by adding the sidebar button to it.  Clicking the sidebar button in my newly modified toolbar caused . . . the VDH sidebar to toggle open & closed.  That is not what it says would happen on that page of documentation.  Next, I did Ctrl+b to open the built-in bookmarks sidebar.  Then I thought to click mouse button 2 on the sidebar button.  Not mouse button 1, which I had been doing, but mouse button 2 to pop up the context menu for the sidebar button.  And look what I found.

![#01](https://github.com/aclap-dev/video-downloadhelper/assets/75588964/ba356457-b77b-4977-a99c-c556e098e1c1)

Is this the answer to your problem?

With the bookmarks sidebar displayed, the sidebar button started toggling the bookmarks sidebar open & closed.  No VDH sidebar in sight.  Well, that's how it's supposed to work.

I fooled around with it some more.  With the bookmarks sidebar still open, I clicked the little drop-down control beside the word `Bookmarks`.  It showed me this:

![#02](https://github.com/aclap-dev/video-downloadhelper/assets/75588964/2f77347e-34c4-4c51-8f9f-77047d6729ed)

The top 3 entries on that menu are the 3 built-in sidebars.  But I decided to click the VDH entry, just for grins & giggles.  Not surprisingly, the VDH sidebar showed up.  But the sidebar button went back to toggling the VDH sidebar, ignoring the built-in sidebars.  Uh . . . what?

Then I noticed there was no little drop-down control beside the words `Video DownloadHelper` at the top of the VDH sidebar.  I thought that was strange.  Then I thought to drag the right edge of the VDH sidebar to the right.

![#03](https://github.com/aclap-dev/video-downloadhelper/assets/75588964/fe55af44-e41f-4008-bd11-edbb4de4c41b)

Now things made some sense.  Let's drop down the menu off this:

![#04](https://github.com/aclap-dev/video-downloadhelper/assets/75588964/3ddcadd0-b8a5-4a4d-a5d5-4a5a069ebc47)

Not surprising.  It looks the same as it did when I had the bookmarks sidebar open.  Alright.  Let's pop up the context menu on the sidebar button now that I have the VDH sidebar showing.

![#05](https://github.com/aclap-dev/video-downloadhelper/assets/75588964/95a3a3b1-50fb-497b-a5f1-990cd02d32c2)

OK.  This is good.  Next, I hit Ctrl+b to replace the VDH sidebar with the built-in bookmarks sidebar.  The sidebar button went back to toggling the bookmarks sidebar open & closed.

So I guess this is how the sidebar feature works.  You can select a sidebar, either a built-in one or a user-supplied one.  The user in this case is, of course, VDH.  Whichever one you select, that becomes the one that gets toggled by the sidebar button.  However, the setting for `Always Show/Only Show in New Tab/Never Show` is visible ONLY if you put the sidebar button on your toolbar, something that is NOT A DEFAULT CUSTOMIZATION.  Notice that in this paragraph I am writing here I am not discussing VDH features.  I am discussing Firefox features.  This business of always showing a sidebar is a Firefox feature, & it's damn hard to find it.  I even went into `about:config` & searched on both `sidebar` & `bookmark` & I didn't find anything that seemed connected to this issue.

So I think I have uncovered what I consider a rather poor user interface design for a feature of Firefox.  Firefox.  Not VDH.  Firefox.  VDH is pretty much just the victim here.  What we've been stumbling over is an exceptionally well-hidden feature of the Firefox UI.

So let us know if clicking that `Never Show` entry on that context menu makes any difference for you.

**Reply by Eldon-Tyrell** [reply]
@25233Guyver

> Press the button at the bottom of the sidebar, the one I showed in the previous image. Closing it using the X button will only close it but not prevent it from happening again.

I did that - but nothing changed with my problem . . .

**Reply by Eldon-Tyrell** [reply]
@GrampaWildWilly

>Let me get this straight.
. . .  
>Do I have this right?

yes exacltly what i did - with only one Firefox window open.


Thank you very much for your investigation effort.
I played around with your findings without success.

you definitely mixed up two things here. The 'Never Show'-setting refers to the Bookmark-TOOLBAR - not the SIDEBAR
and you can find that menu-entry under 
       Main Menu / View / Toolbars / Bookmark-Toolbar
as well.

I fear I will have to give up at some point - although I'd gladly invest more time into that if I only had some direction to go.
totally resetting Firefox as a last resort would not be worth it,

atm I think it is some Firefox bug.

**Reply by GrampaWildWilly** [reply]
@Eldon-Tyrell 
> you definitely mixed up two things here. The 'Never Show'-setting refers to the Bookmark-TOOLBAR - not the SIDEBAR

No no.  Look at my images again.  I'm referring to the sidebar.  Definitely the sidebar.  Not the toolbar.  Look again.

GitHub tends to show images a bit small.  Just click any image.  GitHub will open the image in its own browser window.  You can then magnify the image using the browser's own Ctrl-+ keyboard shortcut.  That may clarify what I've been talking about.

**Reply by GrampaWildWilly** [reply]
Here's my Firefox window.

![#01](https://github.com/aclap-dev/video-downloadhelper/assets/75588964/cc90a805-6bbc-4bd8-962c-cf429ed3bcc8)

I don't have a bookmarks toolbar enabled.  Now look at the sidebar submenu.

![#02](https://github.com/aclap-dev/video-downloadhelper/assets/75588964/2a75ac9c-51e9-4d48-b758-c9d9e5df1273)

There is no cascade submenu for sidebar where you can say `Always Show` or `Never Show`.  The toolbar & sidebar menus are not similar, not symmetrical.  You have to customize your toolbar by adding the sidebar control.  The `Always Show` & `Never Show` menu options are in the context menu for the sidebar control that you have to add to the toolbar.  I didn't pop up a context menu on the toolbar.  I popped up a context menu on a control that I added to the toolbar.  Look at my images again.  Read everything I said as if you've never seen it before.  Don't scan it or skim it.  Read it slowly.

**Reply by GrampaWildWilly** [reply]
Of course, as soon as I mention it, it starts acting up.  This happens every so often.  I'm clicking my images & they are not opening in a new window.  The new window opens but it stays blank.  I don't know why GitHub does this.  I can't even do an `Open Image in New Tab`.  I can copy/paste the image into my image viewer application.  That lets me magnify the images.  This is so frustrating.  Sometimes GitHub works properly, sometimes it doesn't.  If you come back in a few hours, you will be able to click an image & get it to open in a new window.  Consistency.  Where is it when you most need it?

**Reply by GrampaWildWilly** [reply]
Wait.  You're right.  The sidebar control on the toolbar has a popup context menu that talks about the bookmarks toolbar.  That's weird.  Why wouldn't it talk about sidebar properties?  It's a sidebar control.  It should talk about sidebar properties.  OK.  It's completely weird.  I don't have a solution after all.

**Reply by GrampaWildWilly** [reply]
Every button on the toolbar has a popup context menu with a bookmarks entry.  Every button.  What's up with that?  The bookmarks aren't any more important than anything else.  Why isn't there also a sidebar entry on every popup context menu?  This is ridiculous.

**Reply by GrampaWildWilly** [reply]
On the other hand, I found that customizing the toolbar to add the sidebar button & then clicking the added button did successfully toggle a sidebar -- whichever one I most recently selected -- open & closed.  Maybe if you use that button to close your sidebar, your next Firefox session won't have the sidebar.  Maybe.  I find it totally weird that I'm not seeing this problem.  You'd think if there were a bug in Firefox, everybody would experience it.  Isn't software supposed to be deterministic?

**Reply by GrampaWildWilly** [reply]
Maybe it will make a difference if you switch from bookmarks to history, then toggle that closed via the sidebar button, not the X, before you shut down.  Maybe switch to the VDH sidebar, then don't click the sidebar control on the toolbar but click the `Show in popup` button instead.  Have you already tried these?  I'm losing track.

**Comment by Eldon-Tyrell** [comment]
worth a try - but i have tried that and all other combinations in every order for quite some time

thanks anyway - I couldn't imagine how anyone could try to be more helpful than you here !

**Reply by GrampaWildWilly** [reply]
I'm trying but I'm not succeeding.  Maybe this is the bug that Paul mentioned upthread here on 2024/6/8.  I read the discussion over on phabricator & I admit I didn't follow it at all.  But I trust Paul.  So we can only hope a future revision of Firefox, one that appears soon, will fix your problem.

**Comment by ThomasBecker42** [comment]
I have the same problem, no matter if I close the sidebar with the "x" or by the video-downloadhelper icon, after closing Firefox and opening it again, the side bar is open. 
Firefiox 127.0.2 (64-Bit)
Version 9.0.2.12 (stable), target mozilla, locale de.

**Comment by GrampaWildWilly** [comment]
I have just added a comment to #1192 that is relevant here.

**Comment by caddytan** [comment]
This is how I resolve this issue on my Mac.

In Firefox, go to Help --> More Troubleshooting Information
Scroll down to Profile Folder and click on Show In Finder. It will bring up Finder in the Profiles folder
Quit Firefox
In the profiles folder, delete xulstore.json
start Firefox and sidebar is not started automatically.

**Reply by FunnyDrew** [reply]
Thank you! it was helpful on Ubuntu for me!

**Reply by Eldon-Tyrell** [reply]
doing this on PC resulted in the sidebar being always closed on start

**Comment by Eldon-Tyrell** [comment]
I gave up on this and reinstalled Firefox - Then I found something on the web:

The open/closed-status of the sidebar as well as its width are only remebered, if the browser-history is set to being remembered - duh.

Extras / Settings / Security & History / History / "never remember history" --> must be unchecked

I think somebody commented this will be solved in FF-Version 133

I cannot tell if this would have solved my issue here since I already reinstalled Firefox - but it is true at the moment - i have just tested it.

**Comment by WILF219** [comment]
i realise this is an old thread,but i spent a long time looking for an answer,so i'ts worthwhile getting the solution written down
when the sidebar is open,there is a  button at the bottom  called 'show in popup'    that returns  the app to original state

**Reply by mlopezgva** [reply]
> i realise this is an old thread,but i spent a long time looking for an answer,so i'ts worthwhile getting the solution written down when the sidebar is open,there is a button at the bottom called 'show in popup' that returns the app to original state

Yes!

It was easier when it said "Old Interface". I spent more than 2' figuring this out. AND, when clicking on that, I think it would be wise to simply remove the Sidebar entry, just to make it clear what it is doing. IMHO, of course. :-)
